UPGRADE_from_v5_to_v6.md

```diff
@@ -0,0 +1,207 @@
+Registry secret, like this:
+```bash
+export MENDER_REGISTRY_USERNAME="replace-with-your-username"
+export MENDER_REGISTRY_PASSWORD="replace-with-your-password"
+export MENDER_REGISTRY_EMAIL="replace-with-your-email"
+
+kubectl create secret docker-registry my-mender-pull-secret \
+  --docker-username=${MENDER_REGISTRY_USERNAME} \
+  --docker-password=${MENDER_REGISTRY_PASSWORD} \
+  --docker-email=${MENDER_REGISTRY_EMAIL} \
+  --docker-server=registry.mender.io
+```
+and reference it in the `mender-values.yaml` file:
+
+```yaml
+global:
+  # image:
+  #   username: "redacted"
+  #   password: "redacted"
+default:
+  imagePullSecrets:
+    - name: my-mender-pull-secret
+```
+
+### Rootless gui container
+For improved security, the `gui` container is now rootles; this means that
+the `gui.httpPort` is switched from `80` to the unprivileged `8090` port.
+Make sure you are not overriding the `gui.httpPort` in your
+`mender-values.yaml` file.
+
+### Service Keys automatically generated
+You can choose to specify the `device_auth.certs.key`,
+`useradm.certs.key`, and `tenantadm.certs.key` keys in
+the `mender-values.yaml` file, but it is not mandatory anymore.
+If you don't specify them, the Helm Chart will generate them for you.
+
+### NATS and MongoDB subcharts enabled by default
+To facilitate the first Helm Chart installation, the MongoDB and NATS
+subcharts are enabled by default:
+```yaml
+mongodb:
+  enabled: true
+
+nats:
+  enabled: true
+```
+If you are using an external MongoDB or NATS instance, please make sure to
+explicitly disable them before the upgrade:
+```yaml
+mongodb:
+  enabled: false
+
+nats:
+  enabled: false
+```
+
+### Redis subchart disabled by default
+The Redis subchart is disabled by default, because not used in the Open
+Source version. If you want to use it in the Enterprise version, please
+make sure to enable it:
+```yaml
+redis:
+  enabled: true
+```
+
+### Enterprise false by default
+The Enterprise version is disabled by default. If you are using it, please
+make sure to enable it:
+```yaml
+global:
+  enterprise: true
+```
+
+### Storage Proxy enabled by default
+By default, the Storage Proxy is enabled. If you don't want to use it, please
+restore the previous configuration:
+```yaml
+api_gateway:
+  storage_proxy:
+    enabled: false
+
+deployments:
+  customEnvs: []
+```
+
+But if you want to use it, you have to set the `DEPLOYMENTS_STORAGE_PROXY_URI`
+environment variable to the URL exposed, the bucket name in the
+API Gateway configuration, and the `s3.AWS_URI` to the storage endpoint:
+```yaml
+global:
+  s3:
+    AWS_URI: "${STORAGE_ENDPOINT}"
+    AWS_FORCE_PATH_STYLE: "true"
+
+api_gateway:
+  storage_proxy:
+    enabled: true
+    url: "${STORAGE_ENDPOINT}"
+    customRule: "PathRegexp(\`^/${STORAGE_BUCKET}\`)"
+    passHostHeader: false
+
+deployments:
+  customEnvs:
+    - name: DEPLOYMENTS_STORAGE_PROXY_URI
+      value: "https://${MENDER_SERVER_DOMAIN}"
```

Comment:
Should we make it more explicit for lazy people to see an error on unset variables? :)
```suggestion
global:
  s3:
    AWS_URI: "${STORAGE_ENDPOINT:?must be set to your bucket endpoint URL}"
    AWS_FORCE_PATH_STYLE: "true"

api_gateway:
  storage_proxy:
    enabled: true
    url: "${STORAGE_ENDPOINT}"
    customRule: "PathRegexp(\`^/${STORAGE_BUCKET:?must be set to your bucket name}\`)"
    passHostHeader: false

deployments:
  customEnvs:
    - name: DEPLOYMENTS_STORAGE_PROXY_URI
      value: "https://${MENDER_SERVER_DOMAIN:?must be set to your server domain}"
```